### PR TITLE
Donut chart test case modified.

### PR DESCRIPTION
**Pull request by v-asurse**
Modified Donut Chart test cases and resolved some errors.

### CODE REVIEW

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx [hunk 1]
```diff
@@ -92,20 +92,21 @@ export const Arc: React.FunctionComponent<IArcProps> = React.forwardRef<HTMLDivE
...
-    // const id = props.uniqText! + props.data!.data.legend!.replace(/\s+/, '') + props.data!.data.data;
-    const id = 'abcd';
+    const id = props.uniqText! + props.data!.data.legend!.replace(/\s+/, '') + props.data!.data.data;
     const opacity: number = props.activeArc === props.data!.data.legend || props.activeArc === '' ? 1 : 0.1;
     return (
       <g ref={currentRef}>
         {!!focusedArcId && focusedArcId === id && (
           // TODO innerradius and outerradius were absent
           <path
+            id={id + 'focusRing'}
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx:102]: why this change

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.test.tsx [hunk 2]
```diff
@@ -6,14 +6,14 @@ import { resetIds } from '../../Utilities';
 import * as renderer from 'react-test-renderer';
 import { mount, ReactWrapper } from 'enzyme';
 import { IDonutChartProps, DonutChart } from './index';
-import { IDonutChartState, DonutChartBase } from './DonutChart.base';
+//import { IDonutChartState, DonutChartBase } from './DonutChart.base';
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.test.tsx:9]: remove this line

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx [hunk 3]
```diff
@@ -325,16 +322,18 @@ export const DonutChart: React.FunctionComponent<IDonutChartProps> = React.forwa
...
-          <PopoverSurface tabIndex={-1}>
-            <ChartHoverCard
-              Legend={xCalloutValue ? xCalloutValue : legend}
-              YValue={yCalloutValue ? yCalloutValue : value}
-              color={color}
-              culture={props.culture}
-            />
-          </PopoverSurface>
-        </Popover>
+        <div id={_calloutId}>
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx:325]: can you assign the id directly to popover

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChartRTL.test.tsx [hunk 4]
```diff
@@ -27,15 +28,14 @@ describe('Donut chart interactions', () => {
     const { container } = render(
       <DonutChart data={chartPointsDC} innerRadius={55} calloutProps={{ doNotLayer: true }} />,
     );
-
+    const getByClass = queryAllByAttribute.bind(null, 'class');
     // Act
     const getById = queryAllByAttribute.bind(null, 'id');
     fireEvent.mouseOver(getById(container, /Pie/i)[0]);
-    expect(getById(container, /callout/i)[0]).toBeDefined();
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/DonutChartRTL.test.tsx:34]: why remove get by class?

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx (hunk 1 above)
Comment (v-asurse) [reply packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx:102]: These changes provided by sreetama, these id's we are using in some of the testcases.

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.test.tsx (hunk 2 above)
Comment (v-asurse) [reply packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.test.tsx:9]: Removed this line.

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx (hunk 3 above)
Comment (v-asurse) [reply packages/react-components/react-charts-preview/src/components/DonutChart/DonutChart.tsx:325]: Previously, the callout element was using an id, but presently the popover element operates without an id, resulting in the failure of a particular testcase. Now I have removed this id and changed the testcase.

File: packages/react-components/react-charts-preview/src/components/DonutChart/DonutChartRTL.test.tsx (hunk 4 above)
Comment (v-asurse) [reply packages/react-components/react-charts-preview/src/components/DonutChart/DonutChartRTL.test.tsx:34]: After removing the callout id, there's no id present on the DOM itself to locate that particular element, so I resolve this using the getByClass method to locate the element.

File: packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx [hunk 5]
```diff
@@ -92,20 +92,21 @@ export const Arc: React.FunctionComponent<IArcProps> = React.forwardRef<HTMLDivE
 
     const { href, focusedArcId } = props;
     //TO DO 'replace' is throwing error
-    // const id = props.uniqText! + props.data!.data.legend!.replace(/\s+/, '') + props.data!.data.data;
-    const id = 'abcd';
+    const id = props.uniqText! + props.data!.data.legend!.replace(/\s+/, '') + props.data!.data.data;
```
Comment (AtishayMsft) [inline packages/react-components/react-charts-preview/src/components/DonutChart/Arc/Arc.tsx:95]: use useId

**Review by AtishayMsft — APPROVED** [review]

**Comment by AtishayMsft** [comment]
Could you resolve the merge conflict that is coming in DonutChart

**Comment by v-asurse** [comment]
> Could you resolve the merge conflict that is coming in DonutChart

Yes, I resolved the merge conflicts.

**Comment by AtishayMsft** [comment]
There is another merge conflict coming now.
